### PR TITLE
Wip/hughsie/query bound tokenize

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/xb-self-test.c [hunk 1]
```diff
@@ -1719,6 +1719,79 @@ xb_xpath_func(void)
...
+	g_assert_no_error(error);
+	g_assert_nonnull(str);
+	g_debug("\n%s", str);
+
+	/* get component node */
+	cpt_node = xb_silo_query_first(silo, "component", &error);
+	g_assert_no_error(error);
+	g_assert_nonnull(cpt_node);
+
+	/* search using the slow, token-less path (works) */
```
Comment (ximion) [inline src/xb-self-test.c:1763]: We should probably drop the (works) from my original example ;-)

File: src/xb-self-test.c [hunk 2]
```diff
@@ -1719,6 +1719,79 @@ xb_xpath_func(void)
...
+	g_assert_no_error(error);
+	g_assert_nonnull(result);
+
+	n = XB_NODE(g_ptr_array_index(result, 0));
+	g_assert_nonnull(n);
+	g_assert_cmpstr(xb_node_get_text(n), ==, "A strategy game");
+	g_clear_object(&query);
+	g_ptr_array_unref(g_steal_pointer(&result));
+
+	/* search for token (fails) */
```
Comment (ximion) [inline src/xb-self-test.c:1779]: ...same for the (fails) - all of this should be working now.

**Review by ximion — APPROVED** [review]
Looks good to me and makes perfect sense - just maybe drop the works/fails from my original comment strings before merging.

Thank you for looking into this!

File: src/xb-self-test.c (hunk 1 above)
Comment (hughsie) [reply src/xb-self-test.c:1763]: Done.

File: src/xb-self-test.c (hunk 2 above)
Comment (hughsie) [reply src/xb-self-test.c:1779]: Fixed.

**Pull request merged**